specs/spec.md

```diff
@@ -407,8 +413,81 @@ tiger db save-password svc-12345 --password your-password --password-storage key
+
+**Password Behavior:**
+By default, if neither `--password` flag nor `TIGER_NEW_PASSWORD` environment variable is provided, a secure random password will be automatically generated. The password is saved according to the `--password-storage` setting (keyring, pgpass, or none).
+
+**Options for `create role`:**
+- `--name`: Role name to create (required)
+- `--read-only`: Enable permanent read-only enforcement via `tsdb_admin.read_only_role` setting
+- `--from`: Comma-separated list of roles to inherit grants from (e.g., `app_role,readonly_role`)
+- `--statement-timeout`: Set statement timeout for the role (accepts any duration format: "30s", "5m", etc.)
+- `--password`: Password for the role. If provided without a value, prompts interactively. If not provided, checks `TIGER_NEW_PASSWORD` environment variable, otherwise auto-generates a secure random password.
+
+**Read-Only Mode for AI Agents:**
+The `--read-only` flag enables permanent read-only enforcement at the PostgreSQL level using the `tsdb_admin.read_only_role` extension setting. This is designed to provide safe database access for AI agents and automated tools that need to read production data without risk of modification.
+
+**How Read-Only Enforcement Works:**
+- **Role-based (permanent)**: When `--read-only` is set, the role is configured with `ALTER ROLE <role_name> SET tsdb_admin.read_only_role = true`. This locks `transaction_read_only` to always be on for that role, preventing any write operations regardless of other grants or privileges.
+- **Cannot be disabled**: Once set, the read-only state cannot be changed by the role itself, even with `SET transaction_read_only = false`. Only a superuser can modify the role settings.
+- **Survives reconnection**: The read-only enforcement persists across all connections, connection poolers, and database forks.
+
+**Use Cases:**
+- **AI data analysis**: Give AI agents like Claude Code safe access to production data for analysis without risk of accidental modifications
+- **Schema discovery**: Allow AI coding assistants to explore database structure to write better migrations, without write access
+- **Automated reporting**: Run analytics queries against current production data without forking
+- **Development tools**: Provide read-only access to tools that need to inspect but not modify data
+
+**Combining with `--from` for Safe Table Access:**
+The `--from` flag allows you to create a read-only role that inherits the same table access as your application role:
```

Comment:
What is the default behavior if `--from` is not specified? What permissions does the new role have in that case?